react/week2/src/App.js

```diff
@@ -0,0 +1,114 @@
+import "./App.css";
+import { useState, useEffect } from "react";
+
+function TodoItem({ id, description, onDelete }) {
+  const [isChecked, setChecked] = useState(false);
+
+  const check = () => {
+    setChecked(!isChecked);
+  };
+
+  return (
+    <li style={{ 'textDecoration': isChecked ? 'line-through' : 'none' }}>
+      {id}- {description}
+      <input type="checkbox" onClick={check}></input>
+      <button onClick={() => onDelete(id)}>Delete</button>
+    </li>
+  );
+}
+
+function TodoList({ todos }) {
+  const [Todos, setTodos] = useState(todos);
```

Comment:
To avoid the name clashing of "todos" here, you could instead move state up a level to the App function and the pass the the todos and the setTodos as props.

Reply:
Thanks, done.